sympy/sets/fancysets.py

```diff
@@ -352,112 +386,88 @@ def _is_multivariate(self):
         return len(self.lamda.variables) > 1
 
     def _contains(self, other):
-        from sympy.matrices import Matrix
+        from sympy.core.logic import fuzzy_and
         from sympy.solvers.solveset import solveset, linsolve
-        from sympy.solvers.solvers import solve
-        from sympy.utilities.iterables import is_sequence, cartes
-        L = self.lamda
-        if is_sequence(other) != is_sequence(L.expr):
-            return False
-        elif is_sequence(other) and len(L.expr) != len(other):
+
+        other = _sympify(other)
+        expr = self.lamda.expr
+        sig = self.lamda.signature
+        variables = self.lamda.variables
+
+        rep = {v: Dummy(v.name) for v in variables}
+        variables = [v.subs(rep) for v in variables]
+        sig = sig.subs(rep)
+        expr = expr.subs(rep)
+
+        # Map the parts of other to those in the Lambda expr
+        equations = []
+
+        def rmap(ex, ot):
+            if isinstance(ot, Tuple):
+                if not isinstance(ex, Tuple) or len(ex) != len(ot):
+                    return False
+                return all(rmap(e, o) for e, o in zip(ex, ot))
+            else:
+                eq = Eq(ex, ot)
+                if eq is S.false:
+                    return False
+                equations.append(eq)
+                return True
+
+        if not rmap(expr, other):
             return False
 
-        if self._is_multivariate():
-            if not is_sequence(L.expr):
-                # exprs -> (numer, denom) and check again
-                # XXX this is a bad idea -- make the user
-                # remap self to desired form
-                return other.as_numer_denom() in self.func(
-                    Lambda(L.signature, L.expr.as_numer_denom()), self.base_set)
-            eqs = [expr - val for val, expr in zip(other, L.expr)]
-            variables = L.variables
-            free = set(variables)
-            if all(i.is_number for i in list(Matrix(eqs).jacobian(variables))):
-                solns = list(linsolve([e - val for e, val in
-                zip(L.expr, other)], variables))
+        # Map the symbols in the signature to the corresponding domains
+        symsetmap = {}
+
+        def rmatch(sig, bs):
```

Comment:
avoid recursion here again. See a previous comment on the subject.

Reply:
Deep recursion isn't needed here